### PR TITLE
Sync layer stable toolbox

### PR DESCRIPTION
**Pull request by StanislavBreadless**
# What ❔

Set of toolbox/consensus registry changes to make it work with the server + make things more consistent.

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: l1-contracts/contracts/transactionFilterer/GatewayTransactionFilterer.sol [hunk 1]
```diff
@@ -82,18 +83,28 @@ contract GatewayTransactionFilterer is ITransactionFilterer, ReentrancyGuard, Ow
...
+            if (IL2AssetRouter.setAssetHandlerAddress.selector == l2TxSelector) {
+                (, bytes32 decodedAssetId, ) = abi.decode(l2Calldata[4:], (uint256, bytes32, address));
+                return _checkSTMAssetId(decodedAssetId);
+            }
+
             if (
-                (IAssetRouterBase.finalizeDeposit.selector != l2TxSelector) &&
-                (IL2Bridge.finalizeDeposit.selector != l2TxSelector)
+                IAssetRouterBase.finalizeDeposit.selector != l2TxSelector &&
+                IL2Bridge.finalizeDeposit.selector != l2TxSelector
```
Comment (kelemeno) [inline l1-contracts/contracts/transactionFilterer/GatewayTransactionFilterer.sol:94]: Does this allow the chains to be withdrawn btw? We should also allow the withdraw function selector

File: l1-contracts/deploy-scripts/DeployL1.s.sol [hunk 2]
```diff
@@ -41,12 +41,32 @@ import {IL1AssetRouter} from "contracts/bridge/asset-router/IL1AssetRouter.sol";
...
 import {IL1NativeTokenVault} from "contracts/bridge/ntv/IL1NativeTokenVault.sol";
-
+import {L1NullifierDev} from "contracts/dev-contracts/L1NullifierDev.sol";
+import {AccessControlRestriction} from "contracts/governance/AccessControlRestriction.sol";
 import {ICTMDeploymentTracker} from "contracts/bridgehub/ICTMDeploymentTracker.sol";
 import {IMessageRoot} from "contracts/bridgehub/IMessageRoot.sol";
 import {IAssetRouterBase} from "contracts/bridge/asset-router/IAssetRouterBase.sol";
+import {L2ContractsBytecodesLib} from "./L2ContractsBytecodesLib.sol";
+
+struct FixedForceDeploymentsData {
```
Comment (kelemeno) [inline l1-contracts/deploy-scripts/DeployL1.s.sol:58]: We changed the name, but more important you can import the struct

File: l1-contracts/contracts/transactionFilterer/GatewayTransactionFilterer.sol (hunk 1 above)
Comment (StanislavBreadless) [reply l1-contracts/contracts/transactionFilterer/GatewayTransactionFilterer.sol:94]: Let's move it to a different scope, I would want to implement only together with testing for it

Comment (StanislavBreadless) [reply l1-contracts/contracts/transactionFilterer/GatewayTransactionFilterer.sol:94]: https://linear.app/matterlabs/issue/EVM-750/support-chain-withdrawals-from-gateway

File: l1-contracts/deploy-scripts/DeployL1.s.sol (hunk 2 above)
Comment (StanislavBreadless) [reply l1-contracts/deploy-scripts/DeployL1.s.sol:58]: I can not because the original struct is in system contracts

Comment (kelemeno) [reply l1-contracts/deploy-scripts/DeployL1.s.sol:58]: My mistake on both then

File: l1-contracts/test/foundry/integration/deploy-scripts/script-out/output-deploy-hyperchain-10.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+owner_address = "0x70997970C51812dc3A010C7d01b50e0d17dc79C8"
+
+[chain]
+base_token_addr = "0x0000000000000000000000000000000000000001"
```
Comment (kelemeno) [inline l1-contracts/test/foundry/integration/deploy-scripts/script-out/output-deploy-hyperchain-10.toml:4]: Is this file needed, maybe we can gitignore it?
